importers/wecdis_importer.py

```diff
@@ -0,0 +1,494 @@
+from datetime import datetime
+from enum import Enum
+
+from pepys_import.core.formats import unit_registry
+from pepys_import.core.formats.location import Location
+from pepys_import.core.validators import constants
+from pepys_import.file.highlighter.support.combine import combine_tokens
+from pepys_import.file.importer import Importer
+from pepys_import.utils.unit_utils import convert_absolute_angle, convert_distance, convert_speed
+
+
+class MsgType(str, Enum):
+    """The supported WECDIS message types"""
+
+    TIME = "DZA"
+    PLATFORM = "VNM"
+    CONTACT = "CONTACT"
+    POSITION = "POS"  # Could be CPOS, POS, POS1, POS2, ... POSN
+    TMA = "TMA"
+    DEPTH = "PDS"
+    TTM = "TTM"  # Could be TTM1, TTM2, TTM3
+
+
+class WecdisImporter(Importer):
+    def __init__(self):
+        super().__init__(
+            name="WECDIS File Format Importer",
+            validation_level=constants.BASIC_LEVEL,
+            short_name="WECDIS Importer",
+            datafile_type="WECDIS",
+        )
+
+        self.platform_name = None
+        self.timestamp = None
+        self.elevation = None
+
+    def can_load_this_type(self, suffix):
+        return suffix.upper() == ".LOG" or suffix.upper() == ".TXT"
+
+    def can_load_this_filename(self, filename):
+        return True
+
+    def can_load_this_header(self, header):
+        return "$POSL" in header
+
+    def can_load_this_file(self, file_contents):
+        # Need to differentiate from general NMEA - so check charts/version available
+        contents_string = " ".join(file_contents[0:100])
+        return "VER" in contents_string and "CHART" in contents_string
```

Comment:
Aah, I suggest we make the same increased constraint as above here: `,VER,` and `,CHART,`.